pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -313,6 +379,34 @@ func checkOpenStackOpts(openstackOpts *OpenStack) error {
 	return checkMetadataSearchOrder(openstackOpts.metadataOpts.SearchOrder)
 }
 
+// NewOpenStack can be used when we need to use a specific secret
+func NewOpenStack(kubeClient clientset.Interface, name string, namespace string) (*OpenStack, error) {
```

Comment:
I'm trying to understand the implications of having lots of instances of the openstack cloud provider, each talking to potentially different openstack backends, or as different tenants, etc. Is there precedent for constructing per-operation cloud provider instances like this? We need to make sure all the components that deal with cloud providers have the correct instance (this PR opens the possibility means they could have an inconsistently constructed instances)

At first glance, it seems like it might be safer to make the specific operations (volume operations, LB operations) that you want sharded by tenant take additional input (like the new config field in the Cinder volume source), rather than using completely different cloud provider instances.

Reply:
i've removed the auth-url and ca-file, so we will talk only to the openstack configured by the administrator. 

So we are left with being able to configure specific tenant/domain/user/password. which is fine for both load balancers and volumes.